static/app/components/core/badge/tag.chonk.tsx

```diff
@@ -0,0 +1,93 @@
+import type {DO_NOT_USE_ChonkTheme} from '@emotion/react';
+
+import type {TagProps} from 'sentry/components/core/badge/tag';
+import {space} from 'sentry/styles/space';
+import {chonkStyled} from 'sentry/utils/theme/theme.chonk';
+import {unreachable} from 'sentry/utils/unreachable';
+
+type TagType = 'default' | 'info' | 'success' | 'warning' | 'danger' | 'promotion';
+
+interface ChonkTagProps extends Omit<TagProps, 'type'> {
+  type: TagType;
+}
+
+export function chonkTagPropMapping(props: TagProps): ChonkTagProps {
+  return {
+    ...props,
+    type:
+      props.type === 'highlight'
+        ? 'info'
+        : props.type === 'error'
+          ? 'danger'
+          : props.type === 'white'
+            ? 'default'
+            : props.type === 'black'
+              ? 'default'
+              : props.type === undefined
+                ? 'default'
+                : props.type,
```

Comment:
```suggestion
  const legacyTypeMapping = {
    highlight: 'info',
    error: 'danger',
    white: 'default',
    black: 'default',
    [undefined]: 'default'
  };
  return {
    ...props,
    type: legacyTypeMapping[props.type] ?? props.type,
```

Reply:
What is this `[undefined]: 'default'` 😂 ?

Reply:
![CleanShot 2025-03-03 at 13 06 43@2x](https://github.com/user-attachments/assets/e57d6fb0-c018-4cb7-b0d4-4eb35000967a)

It's not very happy. I honestly have little preference here because I plan to delete this afterwards

Reply:
if we make `type` optional in `ChonkTagProps` and then destruct it to `’default’` where necessary, we can do this:

```
const legacyTypeMapping: Partial<Record<NonNullable<TagProps['type']>, TagType>> = {
  highlight: 'info',
  error: 'danger',
  white: 'default',
  black: 'default',
};

export function chonkTagPropMapping(props: TagProps): ChonkTagProps {
  return {
    ...props,
    type: props.type && legacyTypeMapping[props.type],
  };
}
```